fix a bug that in the function ngx_http_subrequest, it will make the …

fishgege opened the pull request:
…headers_in.headers  incorrect，and it will cause many problems

//sr->headers_in = r->headers_in; 
This line of code copy the headers  but make the struct  mistake that  sr->headers_in.headers.last  mismatch &sr->headers_in.headers.part， if somewhere use  sr->headers_in.headers.last ，it will cause assert or some other problems。fx：in the function ngx_http_headers_more_assert 
so I update for this.

shoujj commented:
shoujj

mdounin commented:
The change doesn't seem to make sense.

Please note that this repository is unofficial and pull requests have no chances to be merged. Proper way to submit changes to nginx is via the nginx development mailing list, see http://nginx.org/en/docs/contributing_changes.html.

fishgege committed:
fix a bug that in the function ngx_http_subrequest, it will make the headers_in.headers  incorrect，and it will cause many problems

//sr->headers_in = r->headers_in; 
This line of code copy the headers  but make the struct  mistake that  sr->headers_in.headers.last  mismatch &sr->headers_in.headers.part， if somewhere use  sr->headers_in.headers.last ，it will cause assert or some other problems。fx：in the function ngx_http_headers_more_assert 
so I update for this.

cactushydrocodone commented:
Read the repository header again, and think about this pull request.

fishgege commented:
I got a mistake, it should be the last part
